sortandsearch.c

```diff
@@ -0,0 +1,128 @@
+#include <stdio.h>
+//#include <cs50.h>
+#include <stdbool.h>
+
+void sort(int values[], int n);
+bool search(int value, int values[], int n);
+
+int main(void) {
+    int values[4];
+    values[0] = 39;
+    values[1] = 40;
+    values[2] = 41;
+    values[3] = 42;
+    //printf("values before sorting: \n");
+    //int y;
+    //for (y = 0; y < 5; y++) {
+    //    printf("values: %d\n", values[y]);
+    //}
+    
+    sort(values, 4);
+    //printf("values after sorting: \n");
+    //for (y = 0; y < 5; y++) {
+    //    printf("values: %d\n", values[y]);
+    //}
+    
+    bool found = search(42, values, 4);
+    if (found == true) {
+        printf("Found the value\n");
+    }
+    else {
+        printf("Didn't find it: %d\n", found);
+    }
+}
+
+
+
+void sort(int values[], int n)
+{
+    // TODO: implement an O(n^2) sorting algorithm
+    // Let's try selection sort.
+   
+    int i;
+    int j;
+    int place_counter = 0;
+    
+    for (i = 0; i < n; i++) {
+        //Take ith element and compare it to all the other elements
+        for (j = place_counter; j < n; j++) {
+            if (values[i] > values[j]) {
+                //break out of loop
+                j = n;
+            }
+            
+        }
+        if (j == n) {
+            // Swap places between the smallest element and the element currently in the first position
+            int smallest_element = values[i];
+            int swap_element = values[place_counter];
+
+            values[i] = swap_element;
+            values[place_counter] = smallest_element;
+            
+            // have to move to the next value in values.. the value after the recently sorted smallest element.
+            i = place_counter;
+            
+            place_counter += 1;
+            
+
+        }
+    }
+}
+
+bool search(int value, int values[], int n)
+{
+    // TODO: implement a searching algorithm
+    // Return false if n is non-positive
+    if (n < 0) {
+        return false;
+    }
+    
+    // Base case. 
+    if (n == 1) {
+        if (value ==  values[0]) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    
+    // find midpoint
+    int midpoint = values[n/2];
+    if (value == midpoint) {
```

Comment:
If you ignore this case, will it be handled and finish correctly on the next recursive call?